packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -82,25 +94,59 @@ contract BridgePool is Testable {
         bytes32 indexed priceRequestAncillaryDataHash,
         address indexed caller
     );
-    event ProvidedLiquidity(address indexed token, uint256 amount, uint256 lpTokensMinted, address liquidityProvider);
+
+    event LiquidityAdded(address indexed token, uint256 amount, uint256 lpTokensMinted, address liquidityProvider);
+    event LiquidityRemoved(address indexed token, uint256 amount, uint256 lpTokensBurnt, address liquidityProvider);
 
     modifier onlyFromOptimisticOracle() {
         require(msg.sender == address(_getOptimisticOracle()), "Caller must be OptimisticOracle");
         _;
     }
 
-    constructor(address _bridgeAdmin, address _timer) Testable(_timer) {
+    // TODO: should we consider changing the name of the LP token as a function of the l1Token? if so, might not be able
+    // to do this with this contract inheriting from expanded ERC20 or might need this contract to have an instance
+    // of the LPToken.
```

Comment:
Can you explain this in more detail? I don't quite follow.

Reply:
After this PR, the `BridgePool` _is_ an `ExpandedERC20` token (i.e the pool is a token, to mint the LP tokens). 

The issue with this is that we can dynamically pull out any information from `_l1Token` fed into this contract at deployment time (such as the tokens name/symbol etc) while passing it into the `ExpandedERC20` constructor. What I think we should do rather is just let the deployer choose the name & symbol of the LP tokens.